content/en/docs/getting-started/reference-application-specification.md

```diff
@@ -0,0 +1,107 @@
+    `rolls`
+  - if `rolls` is set to an invalid input (not a number): status code `400` and
+    `{"status": "error", "message": "Parameter rolls must be a positive integer"}`
+  - if `rolls` is set to a `0` or a negative integer: status code `500` and no
+    JSON output.
+- There may be an optional attribute `player=name` to the `/rolldice` endpoint.
+- The application must output the following log lines:
+  - an INFO-level message for each HTTP request with a status code `<400`
+  - a WARN-level message for each HTTP request with a status code between `400`
+    and `499`, including the message which will be sent in the JSON result
+  - a ERROR-level message for each HTTP request with a status code above `499`
+  - if the optional `player` attribute is set, a DEBUG-level message that outputs
+    the value of `player` and the rolled number
+  - if the optional `player` attribute is not set, a DEBUG-level message that outputs
+    the static value `anonymous player` and the rolled number.
+- The code of the application must be split into two files:
+  - an `app` file that contains the handling of the HTTP requests
+  - a `lib` file that contains the implementation of the roll dice function.
+- Error handling for `rolls` should be split as follows:
+  - the `app` checks if `rolls` is defined, and if not, sets it to `1`.
+  - the `app` only checks if `rolls` is a number. If yes, it calls the function
+    for rolling the dice in `lib`. If no, it does the error handling with a
+    `400` error.
+  - the `lib` checks if `rolls` is a positive number. If no, it throws an
+    exception. `app` is catching the error and sending back the `500` error.
+- The `lib` should have an outer function that does the error handling as
+  described above. The outer function then does the following depending on the
+  value of `rolls`:
+  - `rolls == 1`: Run the inner function once and return the value.
+  - `rolls > 1`: Run a loop which calls the inner function `rolls` time and
+    return the results in an array.
+- The inner function of `lib` is creating the random number between 1 and 6 and
+  returns that value.
+
+## Instrumentation requirements
+
+- If possible, the initialization for the OpenTelemetry SDK should be contained
+  in a separate file and imported within the "app" file. Otherwise, it should be
+  part of the "app" file.
+- Common resource detectors should be loaded in the initialization, e.g. for
+  `process`, `container`, `os`, ...
+- The "lib" file must depend only on the OpenTelemetry API.
+- The `service.*` attributes should be added via environment variables
+  (`OTEL_SERVICE_NAME`, `OTEL_RESOURCE_ATTRIBUTES`).
+- Other `resource detectors` should be added to the initialization of the SDK.
+- For exporting telemetry the application should use an exporter for
+  `stdout`/`console` and `otlp`.
+- There should be an option to enable diagnostic logging for the OpenTelemetry
+  components, ideally via `OTEL_LOG_LEVEL`.
```

Comment:
nit: not sure if `OTEL_LOG_LEVEL` is something that all languages support

Reply:
that's why I wrote "ideally" but I can add a "if supported"